test/h5test.h

```diff
@@ -178,25 +186,21 @@ extern pthread_key_t test_thread_info_key_g;
 #ifdef H5_HAVE_MULTITHREAD
 #define TESTING_2(WHAT)                                                                                      \
     do {                                                                                                     \
-        INCR_RUN_COUNT;                                                                                     \
-        if (!TEST_EXECUTION_THREADED) {                                                                           \
-            TESTING_2_DISPLAY(WHAT);                                                                             \
+        INCR_RUN_COUNT;                                                                                      \
+        if (!TEST_EXECUTION_THREADED) {                                                                      \
+            TESTING_2_DISPLAY(WHAT);                                                                         \
         } else {                                                                                             \
-            /* Store test desc for display after test completion */ \
-            thread_info_t *_tinfo = (thread_info_t*)pthread_getspecific(test_thread_info_key_g);                \
-            assert(_tinfo);                                                                                 \
-            /* TBD - Only need to store this for 1 thread */\
-            _tinfo->test_descriptions[_tinfo->num_tests - 1] = WHAT; \
-        }                                                                                                   \
+            /* Store test desc for display after test completion */                                          \
+            thread_info_t *_tinfo = (thread_info_t*)pthread_getspecific(test_thread_info_key_g);             \
+            assert(_tinfo);                                                                                  \
+            /* TBD - Only need to store this for 1 thread */                                                 \
+            _tinfo->test_descriptions[_tinfo->num_tests - 1] = WHAT;                                         \
+        }                                                                                                    \
     } while (0)
 #else 
 #define TESTING_2(WHAT)                                                                                      \
     do {                                                                                                     \
-        INCR_RUN_COUNT;                                                                                     \
-        if (TEST_EXECUTION_THREADED) {                                                                          \
```

Comment:
Removed this error as we shouldn't be throwing errors for a max threads setting even when the library isn't build with multi-thread. It should be a hint rather than a requirement.